sql/src/main/java/io/crate/protocols/postgres/BatchPortal.java

```diff
@@ -0,0 +1,183 @@
+    private List<List<Object>> batchParams = new ArrayList<>();
+    private List<String> queries = new ArrayList<>();
+    private List<Analysis> analysis= new ArrayList<>();
+    private List<FormatCodes.FormatCode[]> resultFormatCodes = new ArrayList<>();
+    private List<List<? extends DataType>> outputTypes = new ArrayList<>();
+    private List<ResultReceiver> resultReceivers = new ArrayList<>();
+
+    BatchPortal(String name,
+                String query,
+                Analysis analysis,
+                List<? extends DataType> outputTypes,
+                ResultReceiver resultReceiver,
+                List<Object> params,
+                SessionData sessionData) {
+        super(name, sessionData);
+        queries.add(query);
+        this.analysis.add(analysis);
+        this.outputTypes.add(outputTypes);
+        resultReceivers.add(resultReceiver);
+        batchParams.add(params);
+    }
+
+    @Override
+    @Nullable
+    public FormatCodes.FormatCode[] getLastResultFormatCodes() {
+        if (resultFormatCodes.isEmpty()) {
+            return null;
+        } else {
+            return resultFormatCodes.get(resultFormatCodes.size() - 1);
+        }
+    }
+
+    @Override
+    public List<? extends DataType> getLastOutputTypes() {
+        return outputTypes.get(outputTypes.size() - 1);
+    }
+
+    @Override
+    public String getLastQuery() {
+        return queries.get(queries.size() - 1);
+    }
+
+    @Override
+    public Portal bind(String statementName, String query, Statement statement,
+                       List<Object> params, @Nullable FormatCodes.FormatCode[] resultFormatCodes) {
+        queries.add(query);
+        batchParams.add(params);
+        this.resultFormatCodes.add(resultFormatCodes);
+        analysis.add(sessionData.getAnalyzer().analyze(statement,
+            new ParameterContext(getArgs(),
+                EMPTY_BULK_ARGS,
+                sessionData.getDefaultSchema())));
+        return this;
+    }
+
+    @Override
+    public List<Field> describe() {
+        Analysis lastAnalysis = analysis.get(analysis.size() - 1);
+        if (lastAnalysis.rootRelation() == null) {
+            return null;
+        }
+        List<Field> fields = lastAnalysis.rootRelation().fields();
+        outputTypes.add(Symbols.extractTypes(fields));
+        return fields;
+    }
+
+    @Override
+    public void execute(ResultReceiver resultReceiver, int maxRows) {
+        validate(analysis.get(analysis.size() - 1));
+        resultReceivers.add(resultReceiver);
+    }
+
+    @Override
+    public void sync(Planner planner, StatsTables statsTables, CompletionListener listener) {
+        BatchCompletionListener batchCompletionListener = new BatchCompletionListener(analysis.size(), listener,
+            new StatsTablesUpdateListener(sessionData.getJobId(), statsTables));
+        for (int i = 0; i < analysis.size(); i++) {
+            Plan plan = planner.plan(analysis.get(i), UUID.randomUUID(), 0, 0);
```

Comment:
@mfussenegger I'm using a new jobId for each operation. Is this correct? any alternatives?